File: languages/java/src/main/java/com/bloock/sdk/entity/EciesDecrypter.java
package com.bloock.sdk.entity;

import com.bloock.sdk.bridge.proto.EncryptionEntities;
import com.bloock.sdk.bridge.proto.EncryptionEntities.EncryptionAlg;

public class EciesDecrypter implements Decrypter {
  EncryptionAlg alg;
  DecrypterArgs args;

  public EciesDecrypter(String publicKey) {
    this.alg = EncryptionAlg.ECIES;
    this.args = new DecrypterArgs(publicKey);
  }

  @Override
  public EncryptionEntities.Decrypter toProto() {
    return EncryptionEntities.Decrypter.newBuilder()
        .setAlg(this.alg)
        .setArgs(this.args.toProto())
        .build();
  }
}

File: languages/php/Bloock/Entity/EciesDecrypter.php
<?php

namespace Bloock\Entity;

class EciesDecrypter implements Decrypter
{
    public $alg;
    public $args;

    public function __construct(string $publicKey) {
        $this->alg = \Bloock\EncryptionAlg::ECIES;
        $this->args = new DecrypterArgs($publicKey);
    }

    public function toProto(): \Bloock\Decrypter {
        $p = new \Bloock\Decrypter();
        $p->setAlg($this->alg);
        $p->setArgs($this->args->toProto());
        return $p;
    }
}